### PR TITLE
fix(pipeline): Solve pipeline cache memory leak problem

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Solve pipeline cache memory leak problem

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Solve pipeline cache memory leak problem         |
| 🇨🇳 中文    |    解决pipeline cache内存泄露问题          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: conf/pipeline/pipeline.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ clusterinfo: {}
 resourcegc: {}
 dbgc: {}
 task-policy: {}
-reconciler-cache: {}
+pipeline-cache: {}
```
Comment (sfwn) [inline conf/pipeline/pipeline.yaml:40]: `cache` is enough.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: conf/pipeline/pipeline.yaml (hunk 1 above)
Comment (littlejiancc) [reply conf/pipeline/pipeline.yaml]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**